### PR TITLE
fix(http sink): Resolved retry issue

### PR DESCRIPTION
**Pull request by nishantmalhotra13**
- resolved the retry issue for http, elasticsearch and clickhouse sinks

- updated the log format of Warn message, as it was not consistent with earlier versions (This one is just a suggestion, and haven't been tested against other sinks). I think the maintainers are better equipped to make decision for log format. I added a comment on code for reference

How to replicate retry issue:

- Configure vector with `http` sink, and provide a non-existent endpoint.
- The vector is supposed to retry in this case, but instead error is being raised and the log is dropped.

### CODE REVIEW

**Review by bruceg — CHANGES_REQUESTED** [review]
This PR appears to have included some extraneous bits in `Cargo.lock`, `Cargo.toml`, and `docs/reference/releases/0.11.1.cue`. Please drop those changes and rebase this on top of current master so we can test that it doesn't introduce any problems.

I think this change looks to be worth doing on the surface, but I am not seeing how it resolves the retry problem. Could you explain what you see going on here?

**Comment by nishantmalhotra13** [comment]
Hi @bruceg 
Thanks for looking into my PR. My branch is based off `v0.11.1` and I haven't tested this against master branch.
What I found was earlier all the http errors raised inside http sink were of type `hyper::Error` but, it was later wrapped with `HTTPError` using snafu in file `src/http.rs` [commit](https://github.com/timberio/vector/commit/eaaab650d81999aeaf88b03891a83082b31a19ea#diff-0820c0d03ae75b35f0b5bbd8dbc8e4d7ca048d19d60351bef069ac00c236910d)
 But, in `src/sinks/util/retries` which holds common retry logic for all the sinks, if any error happens the error handling logic at line 124 `error.downcast_ref::<L::Error>()` tries to get the error type of the sink which raised the error (the downcast call returns hyper::Error in this case). But, the actual error received was of type `HttpError` (as hyper:Error was wrapped using snafu).
Since, the `HttpError` type doesn't match with any of the cases, it is being handled at line 138, which is generic error handling.
I updated the code for http clickhouse and elasticsearch sinks, so that downcast error call at line 124 returns HttpError, and retry happens.
But, to make it work I had to update line 126 as well to use` %expected` for printing instead of `?expected`

File: src/sinks/util/retries.rs [hunk 1]
```diff
@@ -123,7 +123,7 @@ where
 
                 if let Some(expected) = error.downcast_ref::<L::Error>() {
                     if self.logic.is_retriable_error(expected) {
-                        warn!(message = "Retrying after error.", error = ?expected);
+                        warn!(message = "Retrying after error.", error = %expected);
```
Comment (nishantmalhotra13) [inline src/sinks/util/retries.rs:126]: I verified the changes works fine with master branch as well. The one thing I am leaving for you guys to decide is the warning message format.

`warn!(message = "Retrying after error.", error = ?expected);` will print :
`WARN sink{component_kind="sink" component_name=test_out component_type=http}:request{request_id=0}: vector::sinks::util::retries: Retrying after error. error=CallRequest { source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 61, kind: ConnectionRefused, message: "Connection refused" })) }`

But, `warn!(message = "Retrying after error.", error = %expected);` will print :
`WARN sink{component_kind="sink" component_name=test_out component_type=http}:request{request_id=0}: vector::sinks::util::retries: Retrying after error. error=Failed to make HTTP(S) request: error trying to connect: tcp connect error: Connection refused (os error 61)`

Comment (bruceg) [reply src/sinks/util/retries.rs:126]: We definitely prefer the "display" (non-debug) error when possible, so the latter is preferred.

**Review by bruceg — COMMENTED** [review]
I understand your explanation and agree with the change. This is missing a test to ensure the correct behavior persists. I have a couple of ideas how to test this, but if you want to do it, let me know.

**Comment by bruceg** [comment]
Oops, it looks like this needs to have `cargo fmt` run on it. That should fix the checks.

**Comment by nishantmalhotra13** [comment]
Just ran `cargo fmt`.  Should resolve the linting issue :) 
And, for test case, I would be glad to get your help on that.

**Comment by bruceg** [comment]
Replaced by #6542

**Pull request closed without merging**